### ISSUE TITLE
Request for furnace recipes changes

### ISSUE THREAD

**Issue by fridolin2** [minetest/minetest#7300]
- Feature
 
I was working on my mod and so i have found an interesstong idea for an new feature . I want furnace recipes that one or moreitem van be melted in two or more items

**Comment by paramat** [comment]
Yes good feature, needed for making bronze from tin and copper etc.
Related to https://github.com/minetest/minetest_game/issues/2093

**Comment by 4w** [comment]
I'd say this is not a job for the regular furnace but for an alloy furnace like the one from the Technic mod.

So the engine should provide an API for alloying that can be used by all mods.

**Comment by paramat** [comment]
Folding #7767 into this and renaming to general crafting multi-input/output.
